### PR TITLE
search: update repo with prod changes

### PR DESCRIPTION
**Pull request by ntarocco**
@ludmilamarian these are all the changes in `search_engine`. Do you want to keep all or only the `run_on_slave`?

### CODE REVIEW

File: modules/websearch/lib/search_engine.py [hunk 1]
```diff
@@ -1727,20 +1726,14 @@ def get_synonym_terms(term, kbr_name, match_type, use_memoise=False):
...
     if str(ouput_format[0:3]).isdigit() and len(ouput_format) != 6:
         # asked to print MARC tags, but not enough digits,
         # so let's switch back to HTML brief default
         return 'hb'
-    elif format in CFG_WEBSEARCH_BLACKLISTED_FORMATS:
-        if verbose:
-            write_warning("Selected format is not available through perform_request_search", req=req)
-            # Returning an empty list seems dangerous because you wouldn't know
-            # right away that the list is not supposed to be empty.
-        return 'hb'
```
Comment (ludmilamarian) [inline modules/websearch/lib/search_engine.py]: this should remain.. it looks like it's coming from `legacy`, but we did not deployed it.

File: modules/websearch/lib/search_engine.py [hunk 2]
```diff
@@ -5053,8 +5044,13 @@ def print_record(recID, format='hb', ot='', ln=CFG_SITE_LANG, decompress=zlib.de
...
     can_edit_record = False
-    if check_user_can_edit_record(user_info, recID):
-        can_edit_record = True
+    ##################################
+    # TODO_CDS:                      #
+    # It slows down the search a lot #
+    ##################################
+    # if check_user_can_edit_record(user_info, recID):
+    #     can_edit_record = True
+    ##################################
```
Comment (ludmilamarian) [inline modules/websearch/lib/search_engine.py:5061]: ok to commit this part.

File: modules/websearch/lib/search_engine.py [hunk 3]
```diff
@@ -6608,7 +6604,10 @@ def prs_print_records(kwargs=None, results_final=None, req=None, of=None, cc=Non
                    time.time()-seconds_to_rank_and_print,
                    jrec, rg, rm, cc)
     except Exception:
-        register_exception(alert_admin=True)
+        # Turn off exception reporting for Obelix until
+        # cds-recommendation-0X is ready for prod
+        # register_exception(alert_admin=True)
+        pass
```
Comment (ludmilamarian) [inline modules/websearch/lib/search_engine.py]: we should not commit this one.

File: modules/websearch/lib/search_engine.py [hunk 4]
```diff
@@ -4719,9 +4712,7 @@ def print_records(req, recIDs, jrec=1, rg=CFG_WEBSEARCH_DEF_RECORDS_IN_GROUPS, f
                     link_ln = ''
 
                     if 'ln' in req.form:
-                        link_ln = "?ln={0}".format(
-                            req.form.get('ln', CFG_SITE_LANG)
-                        )
+                        link_ln = "?ln={0}".format(ln)
```
Comment (ludmilamarian) [inline modules/websearch/lib/search_engine.py:4722]: ok for this one

**Review by ludmilamarian — CHANGES_REQUESTED** [review]
some of these should not be committed.

**Comment by ntarocco** [comment]
@ludmilamarian I have applied the changes. Please have a look again.

**Review by ludmilamarian — APPROVED** [review]

**Comment by ludmilamarian** [comment]
Thanks for the changes! Just one small cosmetic comment for the future: the prefix needs to be the name of the module, in our case `WebSearch`

**Pull request merged**